structuredevent-model/src/main/java/com/sequenceiq/cloudbreak/structuredevent/event/cdp/CDPStructuredNotificationEvent.java

```diff
@@ -0,0 +1,72 @@
+package com.sequenceiq.cloudbreak.structuredevent.event.cdp;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.sequenceiq.cloudbreak.structuredevent.event.LdapNotificationDetails;
+import com.sequenceiq.cloudbreak.structuredevent.event.RdsNotificationDetails;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CDPStructuredNotificationEvent extends CDPStructuredEvent {
+
+    private LdapNotificationDetails ldapNotificationDetails;
+
+    private RdsNotificationDetails rdsNotificationDetails;
+
+    private CDPStructuredNotificationDetails notificationDetails;
+
+    public CDPStructuredNotificationEvent() {
+        super(CDPStructuredNotificationEvent.class.getSimpleName());
+    }
+
+    public CDPStructuredNotificationEvent(CDPOperationDetails operation, CDPStructuredNotificationDetails notificationDetails) {
+        super(CDPStructuredNotificationEvent.class.getSimpleName(), operation);
+        this.notificationDetails = notificationDetails;
+        ldapNotificationDetails = null;
+        rdsNotificationDetails = null;
```

Comment:
why these null initializations required? the same for those in the other constructors